samples/drivers/spi_controller/src/main.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025 Toney Abraham
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(spi_controller, LOG_LEVEL_INF);
+
+#define SPI_NODE DT_ALIAS(spi0)
+
+static const struct device *spi_dev = DEVICE_DT_GET(SPI_NODE);
+
+static const struct spi_config spis_config = {
+	.operation = SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | SPI_WORD_SET(8) | SPI_MODE_LOOP,
```

Comment:
SPI_MODE_LOOP might not be supported by all controllers, so this has to come as an option managed by a dedicated Kconfig in your sample (see tests/drivers/spi/spi_loopback/Kconfig )